plugins/ipam/host-local/main.go

```diff
@@ -26,49 +28,99 @@ func main() {
 }
 
 func cmdAdd(args *skel.CmdArgs) error {
-	ipamConf, err := LoadIPAMConfig(args.StdinData, args.Args)
+	ipamConf, ipamConf6, err := LoadIPAMConfig(args.StdinData, args.Args)
 	if err != nil {
 		return err
 	}
 
-	store, err := disk.New(ipamConf.Name)
-	if err != nil {
-		return err
+	r := &types.Result{
+		IP4: nil,
+		IP6: nil,
 	}
-	defer store.Close()
 
-	allocator, err := NewIPAllocator(ipamConf, store)
-	if err != nil {
-		return err
-	}
+	if ipamConf != nil {
+		store, err := disk.New(ipamConf.Name)
+		if err != nil {
+			return err
+		}
+		defer store.Close()
 
-	ipConf, err := allocator.Get(args.ContainerID)
-	if err != nil {
-		return err
+		allocator, err := NewIPAllocator(ipamConf, store)
+		if err != nil {
+			return err
+		}
+
+		ipConf, err := allocator.Get(args.ContainerID)
+		if err != nil {
+			return err
+		}
+		r.IP4 = ipConf
 	}
 
-	r := &types.Result{
-		IP4: ipConf,
+	if ipamConf6 != nil {
+		store, err := disk.New(ipamConf6.Name)
+		if err != nil {
+			return err
+		}
+		defer store.Close()
+
+		allocator, err := NewIPAllocator(ipamConf6, store)
+		if err != nil {
+			return err
+		}
+		ipConf6, err := allocator.Get(args.ContainerID)
+		if err != nil {
+			return err
+		}
+		r.IP6 = ipConf6
 	}
+
 	return r.Print()
 }
 
 func cmdDel(args *skel.CmdArgs) error {
-	ipamConf, err := LoadIPAMConfig(args.StdinData, args.Args)
+	var result [2]error
+	ipamConf, ipamConf6, err := LoadIPAMConfig(args.StdinData, args.Args)
 	if err != nil {
 		return err
 	}
 
-	store, err := disk.New(ipamConf.Name)
-	if err != nil {
-		return err
-	}
-	defer store.Close()
+	if ipamConf != nil {
+		store, err := disk.New(ipamConf.Name)
+		if err != nil {
+			return err
+		}
+		defer store.Close()
 
-	allocator, err := NewIPAllocator(ipamConf, store)
-	if err != nil {
-		return err
+		allocator, err := NewIPAllocator(ipamConf, store)
+		if err != nil {
+			return err
+		}
+
+		e := allocator.Release(args.ContainerID)
+		if e != nil {
+			result[0] = e
+		}
 	}
 
-	return allocator.Release(args.ContainerID)
+	if ipamConf6 != nil {
+		store, err := disk.New(ipamConf6.Name)
+		if err != nil {
+			return err
```

Comment:
I'm arguing that, in this case, you are no longer returning `result[0]` generated on line 102. I think you, again, want to get all the "resource handles" before proceeding with the "transaction".
It's not that the ipv4 won't be released, but that information is being lost.